src/transactions/test/OfferTests.cpp

```diff
@@ -674,350 +674,435 @@ TEST_CASE("create offer", "[tx][offers]")
-                        market.requireChangesWithOffer(
-                            {},
-                            [&] {
-                                return market.addOffer(
-                                    a1, {usd, idr, exactCrossPrice, 150});
-                            }),
-                        ex_MANAGE_SELL_OFFER_CROSS_SELF);
-                });
-            }
+                    auto toSet = static_cast<uint32_t>(AUTH_REQUIRED_FLAG) |
+                                 static_cast<uint32_t>(AUTH_REVOCABLE_FLAG);
 
-            SECTION("offer crosses and removes first")
-            {
-                for_all_versions(*app, [&] {
-                    issuer.pay(b1, usd, 20000);
+                    issuer.setOptions(setFlags(toSet));
+                    Asset const& asset =
+                        authState == AuthState::
+                                         BUY_AUTHORIZED_TO_MAINTAIN_LIABILITIES
+                            ? usd
+                            : idr;
+                    issuer.allowMaintainLiabilities(asset, a1);
+                };
 
-                    // offer is sell 150 USD for 100 USD; sell USD @ 1.5 /
-                    // buy IRD @ 0.66
-                    market.requireChangesWithOffer(
-                        {{offers[0].key, OfferState::DELETED},
-                         {offers[1].key, offers[1].state}},
-                        [&] {
-                            return market.addOffer(
-                                b1, {usd, idr, exactCrossPrice, 150},
-                                OfferState::DELETED);
+                SECTION("offer does not cross")
+                {
+                    for_all_versions(*app, [&] {
+                        a1CheckSetAuthToMaintainLiabilities();
+                        issuer.pay(b1, usd, 20000);
+                        // offer is sell 40 USD for 80 IDR ; sell USD @ 2
+                        market.requireChangesWithOffer({}, [&] {
+                            return market.addOffer(b1,
+                                                   {usd, idr, Price{2, 1}, 40});
                         });
-                });
-            }
-
-            SECTION("offer crosses, removes first six and changes seventh")
-            {
-                for_versions_to(9, *app, [&] {
-                    issuer.pay(b1, usd, 20000);
-
-                    market.requireBalances({{a1, {{usd, 0}, {idr, 100000}}},
-                                            {b1, {{usd, 20000}, {idr, 0}}}});
+                    });
+                }
 
-                    // Offers are: sell 100 IDR for 150 USD; sell IRD @ 0.66
-                    // -> buy USD @ 1.5
-                    // first 6 offers get taken for 6*150=900 USD, gets 600
-                    // IDR in return
+                SECTION("offer crosses own")
+                {
+                    for_all_versions(*app, [&] {
+                        issuer.pay(a1, usd, 20000);
```

Comment:
Missing `a1CheckSet...`?

Reply:
This tests that an account can't add an offer to cross with one of its own existing offers. If we're authorized to maintain liabilities, then the add offer will fail for a different reason (`*_NOT_AUTHORIZED`). We already test that an offer can't be added when an account is in this state in `AllowTrustTests`.